ports/pulsar-client-cpp/unofficial-pulsar-config.cmake

```
@@ -61,6 +56,15 @@ if (NOT TARGET unofficial::pulsar::pulsar)
                 IMPORTED_LOCATION_DEBUG "${_pulsar_library_debug}")
             unset(_pulsar_library_debug CACHE)
         endif ()
+        if (NOT MSVC)
+            if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+                set_target_properties(unofficial::pulsar::pulsar PROPERTIES
+                    IMPORTED_LOCATION "${_pulsar_library_debug}")
+            else ()
+                set_target_properties(unofficial::pulsar::pulsar PROPERTIES
+                    IMPORTED_LOCATION "${_pulsar_library_release}")
+            endif ()
+        endif ()
```

```suggestion
        if (NOT MSVC)
            if (CMAKE_BUILD_TYPE STREQUAL "Debug")
                set_target_properties(unofficial::pulsar::pulsar PROPERTIES
                    IMPORTED_LOCATION "${_pulsar_library_debug}")
            else ()
                set_target_properties(unofficial::pulsar::pulsar PROPERTIES
                    IMPORTED_LOCATION "${_pulsar_library_release}")
            endif ()
        endif ()
```

This does not work for multi config generators. The way to do this by setting `IMPORTED_LOCATION_<CONFIG>`

---

I supported multi-config generators here: https://github.com/microsoft/vcpkg/blob/c9140a3b500812ad3206317885860d9553b93f13/ports/pulsar-client-cpp/unofficial-pulsar-config.cmake#L57-L61

With MSVC generator, setting `IMPORTED_LOCATION_RELEASE` and `IMPORTED_LOCATION_DEBUG` is enough. However, with Unix Makefiles, if `IMPORTED_LOCATION` is not set, there will be the following error when running `cmake --build`

```
*** target pattern contains no `%'.  Stop.
```

---

You need to specify CMAKE_BUILD_TYPE  since you are using a single config generator

---

I figured out the reason. It seems to be caused by the lack of `IMPORTED_CONFIGURATIONS`. After I modified the code above to:

```cmake
        add_library(unofficial::pulsar::pulsar UNKNOWN IMPORTED)
        set_target_properties(unofficial::pulsar::pulsar PROPERTIES
            IMPORTED_CONFIGURATIONS "RELEASE"
            IMPORTED_LOCATION_RELEASE "${_pulsar_library_release}")
        if (_pulsar_library_debug)
            set_target_properties(unofficial::pulsar::pulsar PROPERTIES
                IMPORTED_CONFIGURATIONS "RELEASE;DEBUG"
                IMPORTED_LOCATION_DEBUG "${_pulsar_library_debug}")
            unset(_pulsar_library_debug CACHE)
        endif ()
```

It also worked without specifying `CMAKE_BUILD_TYPE`.